crates/ty_python_semantic/resources/mdtest/assignment/annotations.md

```diff
@@ -88,6 +88,33 @@ def foo(v: str | int | None, w: str | str | None, x: str | str):
     reveal_type(x)  # revealed: str
 ```
 
+## PEP-604 in non-type-expression context
+
+### In Python 3.10 and later
+
+```toml
+[environment]
+python-version = "3.10"
+```
+
+```py
+IntOrStr = int | str
+```
+
+### Earlier versions
```

Comment:
Should there be a test case for stringified annotations (`from __future__ import annotations`)

Reply:
I don't understand how that's related to the functionality being added here? `int | str` in *non-type-expressions* is a error in Python 3.9 and earlier, no matter if annotations are deferred or not, right?

Reply:
if `from __future__ import` is present at the top of the file then all annotations are stringified, and so they should not error even in 3.9.

Reply:
In annotations, yes. But not in non-type expressions: 
```pycon
▶ uv run -p 3.9 python
Python 3.9.20 (main, Oct 16 2024, 04:36:33) 
[Clang 18.1.8 ] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from __future__ import annotations
>>> def f(x: int | str): ...
... 
>>> A = int | str
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for |: 'type' and 'type'
```

Reply:
I think it would be clearer to say "non-annotations" than "non-type-expressions", since (with implicit type alias support, which we don't fully have yet), `int | str` in `A = int | str` should be evaluated as a type expression -- but it will still fail at runtime in older Python versions, regardless of `from __future__ import annotations`, because it is not syntactically an _annotation_, and `from __future__ import annotations` affects only _annotations_.

Reply:
(This will probably also add complexity to the implementation to keep these tests passing when we improve our support for implicit type aliases, because we might need to evaluate these both as value expressions -- in order to get these diagnostics -- and as type expressions -- in order to know how to interpret them when used as a type alias. )

Reply:
I am sorry, I should have read the PR description more closely. The linked issue talks about the union syntax in function definitions so I got confused. I am assuming that error messages coming from type expressions are coming in a separate PR? (ie. to address #astral-sh/ty#449).